test/e2e/e2e_test.go

```diff
@@ -135,3 +137,46 @@ func TestMain(m *testing.M) {
 func TestE2E(t *testing.T) {
 	RunE2ETests(t)
 }
+
+var _ = ginkgo.ReportAfterSuite("Kubernetes e2e suite report", func(report ginkgo.Report) {
+	var f *os.File
+	var err error
+	// The DetailsRepoerter will output details about every test (name, files, lines, etc) which helps
+	// when documenting our tests.
+	if len(framework.TestContext.SpecSummaryOutput) <= 0 {
+		return
+	}
+	if _, err = os.Stat(framework.TestContext.SpecSummaryOutput); os.IsNotExist(err) {
+		f, err = os.Create(framework.TestContext.SpecSummaryOutput)
+		if err != nil {
+			klog.Errorf("Failed to create reporter: %v", err)
+			return
+		}
+	} else {
+		f, err = os.OpenFile(framework.TestContext.SpecSummaryOutput, os.O_RDWR|os.O_APPEND, 0660)
+		if err != nil {
+			klog.Errorf("Failed to open the reporter: %v", err)
+			return
+		}
+	}
+
+	for _, specReport := range report.SpecReports {
+		b, err := specReport.MarshalJSON()
+		if err != nil {
+			klog.Errorf("Error in detail reporter: %v", err)
+			return
+		}
+		_, err = f.Write(b)
+		if err != nil {
+			klog.Errorf("Error saving test details in detail reporter: %v", err)
+			return
+		}
+		// Printing newline between records for easier viewing in various tools.
+		_, err = fmt.Fprintln(f, "")
+		if err != nil {
+			klog.Errorf("Error saving test details in detail reporter: %v", err)
+			return
+		}
+	}
+	f.Close()
```

Comment:
should you defer f.Close() before the for loop , L162, in case some of the steps fail you guarantee the file is closed?